Sources/Core/Client/Channel/Channel+RxEvents.swift

```diff
@@ -21,8 +21,10 @@ public extension Reactive where Base == Channel {
     
     /// Observe all events for this channel.
     var events: Observable<StreamChatClient.Event> {
-        associated(to: base, key: &Channel.rxOnEvent) { [unowned base] in
+        associated(to: base, key: &Channel.rxOnEvent) { [weak base] in
             Observable<StreamChatClient.Event>.create({ observer in
+                guard let base = base else { return Disposables.create() }
```

Comment:
You don't need this, just let `subscription` be optional and call `subscription?.cancel()`

Same for below

Reply:
Fixed.